fe/fe-core/src/main/java/com/starrocks/alter/LakeTableAlterMetaJob.java

```diff
@@ -0,0 +1,497 @@
+import com.starrocks.lake.Utils;
+import com.starrocks.persist.gson.GsonUtils;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.task.AgentBatchTask;
+import com.starrocks.task.AgentTaskExecutor;
+import com.starrocks.task.AgentTaskQueue;
+import com.starrocks.task.UpdateTabletMetaInfoTask;
+import com.starrocks.thrift.TTabletMetaType;
+import com.starrocks.thrift.TTabletType;
+import com.starrocks.thrift.TTaskType;
+import io.opentelemetry.api.trace.StatusCode;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import javax.validation.constraints.NotNull;
+
+public class LakeTableAlterMetaJob extends AlterJobV2 {
+
+    private static final Logger LOG = LogManager.getLogger(LakeTableAlterMetaJob.class);
+
+    @SerializedName(value = "metaType")
+    private TTabletMetaType metaType;
+
+    @SerializedName(value = "metaValue")
+    private boolean metaValue;
+
+    @SerializedName(value = "watershedTxnId")
+    private long watershedTxnId = -1;
+
+    // PartitionId -> indexId -> MaterializedIndex
+    @SerializedName(value = "partitionIndexMap")
+    private Table<Long, Long, MaterializedIndex> partitionIndexMap = HashBasedTable.create();
+
+    @SerializedName(value = "commitVersionMap")
+    // Mapping from partition id to commit version
+    private Map<Long, Long> commitVersionMap = new HashMap<>();
+
+    public LakeTableAlterMetaJob(long jobId, long dbId, long tableId, String tableName,
+                                 long timeoutMs, TTabletMetaType metaType, boolean metaValue) {
+        super(jobId, JobType.SCHEMA_CHANGE, dbId, tableId, tableName, timeoutMs);
+        this.metaType = metaType;
+        this.metaValue = metaValue;
+    }
+
+    @Override
+    protected void runPendingJob() throws AlterCancelException {
+        // send task to be
+        List<Partition> partitions = Lists.newArrayList();
+        OlapTable olapTable;
+        Database db = GlobalStateMgr.getCurrentState().getDb(dbId);
+
+        if (db == null) {
+            throw new AlterCancelException("database does not exist, dbId:" + dbId);
+        }
+        db.readLock();
+        try {
+            olapTable = (OlapTable) db.getTable(tableName);
+            if (olapTable == null) {
+                throw new AlterCancelException("table does not exist, tableName:" + tableName);
+            }
+            partitions.addAll(olapTable.getPartitions());
+        } finally {
+            db.readUnlock();
+        }
+
+        // If the function of updatePartitionTabletMeta has been called,
+        // then FE restart or change leader,
+        // updatePartitionTabletMeta will have the same txnId so that
+        // BE can judge whether the task has been processed。
```

Comment:
This logic has been removed from BE, right? Is it still necessary to persist the txn id now?

Reply:
Yes, I will remove this logic to persist the txn id